### ISSUE TITLE
Pointless mysql query to keep connection alive… doesn't work?

### ISSUE THREAD

**Issue by tbnv** [matomo-org/plugin-GoogleAnalyticsImporter#20]
Hi,

Still getting timeouts and "mysql server has gone away" errors despite the "pointless mysql query to keep connection alive".

```
WARNING [2019-11-13 12:42:31] 88380  /libs/Zend/Db/Statement/Pdo.php(228): Warning - PDOStatement::execute(): MySQL server has gone away - Matomo 3.12.0 - Please report this message in the Matomo forums: https://forum.matomo.org (please do a search first as it might have been reported already)
WARNING [2019-11-13 12:42:31] 88380  /libs/Zend/Db/Statement/Pdo.php(228): Warning - PDOStatement::execute(): Error reading result set's header - Matomo 3.12.0 - Please report this message in the Matomo forums: https://forum.matomo.org (please do a search first as it might have been reported already)
  [Zend_Db_Statement_Exception]                                    
  SQLSTATE[HY000]: General error: 2006 MySQL server has gone away
```

**Comment by diosmosis** [comment]
Well, I had hoped it would. Are you able to diagnose the specific problem with your mysql config (see https://matomo.org/faq/troubleshooting/faq_183/ for possible causes)?

**Comment by tbnv** [comment]
This instance of Matomo is installed on a shared hosting, so I unfortunately don't have access to the `my.cnf` to change these parameters. The hosting provider apparently set `wait_timeout` and `interactive_timeout` to 30 seconds.

In their FAQ they suggest to run a script that regularly pings or queries, test if the connection still exists, and restart it if necessary.

**Comment by diosmosis** [comment]
The query added in the new release is the regular query to keep the connection alive. However, the plugin waits at most a minute in between queries if rate limited, so I suppose there needs to be more pointless queries in your case. I'll be able to add that change today.

**Comment by tbnv** [comment]
Great, thanks.

**Comment by diosmosis** [comment]
@tbnv can you see if these changes work for you: https://github.com/matomo-org/plugin-GoogleAnalyticsImporter/pull/21 ?

**Comment by tbnv** [comment]
Just tested that and it seems it's working.

Running into another problem but I guess this one is on GA's side ("Failed to reach GA after 30 attempts. Restart the import later.").

**Comment by diosmosis** [comment]
You can run the command w/ `-vvv` to get more information in that case. It should show the response from GA when it fails, though that means there will be quite a bit to look through. If the daily limit were reached, the plugin would know, so the error you're getting must be something else.

**Comment by diosmosis** [comment]
Fixed by #21
